2019-10-21 07:19:01,536 ERROR: Exception on /index [GET] [in /home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/home/skullquake/src/skullquake/pythonsamples/src/flsk_mt_006/app/routes.py", line 42, in index
    posts=posts
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/flask/templating.py", line 140, in render_template
    ctx.app,
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/flask/templating.py", line 120, in _render
    rv = template.render(context)
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/home/skullquake/src/skullquake/pythonsamples/src/flsk_mt_006/app/templates/index.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/home/skullquake/src/skullquake/pythonsamples/src/flsk_mt_006/app/templates/base.html", line 56, in top-level template code
    {% block content %}{% endblock %}
  File "/home/skullquake/src/skullquake/pythonsamples/src/flsk_mt_006/app/templates/index.html", line 5, in block "content"
    {{ form.hidden_tag() }}
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/jinja2/environment.py", line 430, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'form' is undefined
2019-10-21 07:21:26,906 INFO: Microblog startup [in /home/skullquake/src/skullquake/pythonsamples/src/flsk_mt_006/app/__init__.py:51]
2019-10-21 07:22:01,669 INFO: Microblog startup [in /home/skullquake/src/skullquake/pythonsamples/src/flsk_mt_006/app/__init__.py:51]
2019-10-21 07:22:02,815 ERROR: Exception on /index [POST] [in /home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/flask/app.py:1891]
Traceback (most recent call last):
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/skullquake/src/skullquake/pythonsamples/src/venv/lib/python3.6/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/home/skullquake/src/skullquake/pythonsamples/src/flsk_mt_006/app/routes.py", line 33, in index
    post = Post(body=form.post.data, author=current_user)
NameError: name 'Post' is not defined
2019-10-21 07:22:36,824 INFO: Microblog startup [in /home/skullquake/src/skullquake/pythonsamples/src/flsk_mt_006/app/__init__.py:51]
2019-10-21 07:23:31,030 INFO: Microblog startup [in /home/skullquake/src/skullquake/pythonsamples/src/flsk_mt_006/app/__init__.py:51]
2019-10-21 09:28:36,904 INFO: Microblog startup [in /home/skullquake/src/skullquake/pythonsamples/src/flsk_mt_006/app/__init__.py:51]
2019-10-21 09:30:27,628 INFO: Microblog startup [in /home/skullquake/src/skullquake/pythonsamples/src/flsk_mt_006/app/__init__.py:51]
2019-10-21 09:32:32,910 INFO: Microblog startup [in /home/skullquake/src/skullquake/pythonsamples/src/flsk_mt_006/app/__init__.py:51]
2019-10-21 09:33:20,319 INFO: Microblog startup [in /home/skullquake/src/skullquake/pythonsamples/src/flsk_mt_006/app/__init__.py:51]
2019-10-21 09:33:45,693 INFO: Microblog startup [in /home/skullquake/src/skullquake/pythonsamples/src/flsk_mt_006/app/__init__.py:51]
2019-10-21 09:35:36,830 INFO: Microblog startup [in /home/skullquake/src/skullquake/pythonsamples/src/flsk_mt_006/app/__init__.py:51]
2019-10-21 09:38:52,503 INFO: Microblog startup [in /home/skullquake/src/skullquake/pythonsamples/src/flsk_mt_006/app/__init__.py:51]
